contracts/stake/src/state.rs

```diff
@@ -214,21 +187,52 @@ impl StakeState {
         self.stakes.get_mut(&pk.to_bytes()).unwrap()
     }
 
-    /// Gets a mutable reference to the stake of a given key.
-    pub(crate) fn load_stake_mut(
-        &mut self,
-        pk: &PublicKey,
-    ) -> Option<&mut StakeData> {
-        self.stakes.get_mut(&pk.to_bytes())
-    }
-
     /// Rewards a `public_key` with the given `value`. If a stake does not exist
     /// in the map for the key one will be created.
     pub fn reward(&mut self, public_key: &PublicKey, value: u64) {
         let stake = self.load_or_create_stake_mut(public_key);
         stake.increase_reward(value);
     }
 
+    /// Total amount slashed from the genesis
+    pub fn slashed_amount(&self) -> u64 {
+        self.slashed_amount
+    }
+
+    /// Slash the given `to_subtract` amount from a `public_key` stake (if
+    /// any). Firstly the amount is subtracted from the reward, if that's
+    /// not enough the stake amount is touched.
+    pub fn slash(&mut self, public_key: &PublicKey, to_slash: u64) {
```

Comment:
@fed-franz Worth to take this into consideration for further DIP consolidation

Reply:
The deflationary slash is intentional and resulting from a discussion between @autholykos @herr-seppia. 
To be honest, I don't recall a very strong motivation behind this choice, but this was indeed the result of other options being discarded.

For instance,  having the slash being `emission/max_iterations` (i.e. 1/255 of the coinbase), which would keep it inflationary in most cases, was also considered (and discarded).

I think the main takeaway of the discussion was that this option would overall compensate with the block emissions because we don't expect many failed iterations.

Honestly, it wouldn't hurt to have a second round of discussion on this.

As for what you mention, having provisioners incentivized to vote for slashing is a big problem, because there's no way to prevent them from voting Timeout (MissingCandidate) even if the candidate has been received.
Moreover, this would add to the implicit incentive provisioners have to make a candidate fail when they are a future-iteration generator (this issue has also been extensively discussed but no Issue has been filed to address it).